stackslib/src/chainstate/stacks/boot/pox-4.clar

```diff
@@ -444,12 +450,13 @@
 ;; A PoX address can be added to at most 12 consecutive cycles.
 ;; No checking is done.
 (define-private (add-pox-partial-stacked (pox-addr (tuple (version (buff 1)) (hashbytes (buff 32))))
+                                         (signer principal)
```

Comment:
Why does the partial stacked state need to know about the signer?

Thinking about pooled stacking, it makes sense to provide the key only on aggregate contract calls. The voting slots are based on aggregate values as well.

Reply:
Yeah -- I agree with this. Partial stacking state shouldn't have the signer, the signer should be set on aggregate-commit. The `stacking-state` map also should not have a `signer-key` field.

So I'd recommend:

1. Removing `signer-key` from `stacking-state`
2. Removing `signer` from the `partial-stacked-by-cycle`
3. Add `signer-key` argument to the `aggregate-commit` method.

Otherwise, the changes in pox-4 look good to me.

Reply:
Copy. Wrote similar thoughts here #4247 just a few mins ago. Assigning to here / @MarvinJanssen.

Reply:
Yep we were going back and forth on it. Will change it soon. Thanks @friedger

Reply:
Edit: is having the signer key in stacking state not useful to have?

Reply:
So is the plan to use `(signer principal)` or `(signer-key (buff 33))`?

Reply:
The arguments to stacking are `(signer-key (buff 33))`, but the internal data maps store `principal`. The stacks functions will need to compute the principal from the signer key.

Reply:
> Yeah -- I agree with this. Partial stacking state shouldn't have the signer, the signer should be set on aggregate-commit. The `stacking-state` map also should not have a `signer-key` field.
> 
> So I'd recommend:
> 
> 1. Removing `signer-key` from `stacking-state`
> 2. Removing `signer` from the `partial-stacked-by-cycle`
> 3. Add `signer-key` argument to the `aggregate-commit` method.
> 
> Otherwise, the changes in pox-4 look good to me.

@kantai Are you going to make this happen?

Reply:
@kantai @MarvinJanssen @hstove per @jferrant, we actually need to store the signing keys to the stacker map after all.